### PR TITLE
My profile page

### PR DESCRIPTION
**Pull request by yashrajoria3**

### CODE REVIEW

File: src/app/(with-sidebar)/myProfile/_hook/useMyProfile.tsx [hunk 1]
```diff
@@ -40,23 +63,68 @@ const useMyProfilePage = (): useMyProfilePage => {
   const [skip, setSkip] = useState(0);
   const [hasMorePosts, setHasMorePosts] = useState<boolean | undefined>(true);
   const [post, setPosts] = useState<ArticlesList[]>([]);
+  const [isEditProfileOpen, setIsEditProfileOpen] = useState(false);
+
+  const [userInfo, setUserInfo] = useState<{
+    name: string;
```
Comment (thekalamanchofficial) [inline src/app/(with-sidebar)/myProfile/_hook/useMyProfile.tsx]: declare a type for it

File: src/app/(with-sidebar)/myProfile/_hook/useMyProfile.tsx [hunk 2]
```diff
@@ -40,23 +63,68 @@ const useMyProfilePage = (): useMyProfilePage => {
...
+      userEmail: user?.primaryEmailAddress?.emailAddress ?? "",
+    },
+    {
+      enabled: !!user?.primaryEmailAddress?.emailAddress,
+    },
+  );
+
+  useEffect(() => {
+    setUserInfo({
+      name: userDetails?.name ?? "",
```
Comment (thekalamanchofficial) [inline src/app/(with-sidebar)/myProfile/_hook/useMyProfile.tsx:60]: why do we need useEffect here? Just pass a prop to MyProfile component and use the prop

File: src/app/(with-sidebar)/myProfile/page.tsx [hunk 3]
```diff
@@ -101,19 +123,50 @@ const MyProfile = () => {
...
       />
       <CustomTabs tabs={tabs} activeTab={tab} onTabChange={handleChange} />
       {renderUI}
+      {isEditProfileOpen && (
+        <EditProfile
+          open={isEditProfileOpen}
+          handleClose={handleEditProfileClose}
+          profileData={userInfo}
+          handleProfileSave={async ({
+            name,
```
Comment (thekalamanchofficial) [inline src/app/(with-sidebar)/myProfile/page.tsx]: use a type here and move all this logic to hook, just expose a callback

**Review by thekalamanchofficial — CHANGES_REQUESTED** [review]
Vercal checks are failing

File: src/app/(with-sidebar)/myProfile/_hook/useMyProfile.tsx (hunk 1 above)
Comment (yashrajoria3) [reply src/app/(with-sidebar)/myProfile/_hook/useMyProfile.tsx]: done

File: src/app/(with-sidebar)/myProfile/page.tsx (hunk 3 above)
Comment (yashrajoria3) [reply src/app/(with-sidebar)/myProfile/page.tsx]: done

File: src/app/(with-sidebar)/myProfile/_hook/useMyProfile.tsx (hunk 2 above)
Comment (yashrajoria3) [reply src/app/(with-sidebar)/myProfile/_hook/useMyProfile.tsx:60]: keeping useEffect for now

**Review by thekalamanchofficial — APPROVED** [review]

**Pull request merged**